plugins/outputs/loki/stream.go

```diff
@@ -0,0 +1,71 @@
+package loki
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+)
+
+type (
+	Log []string
+
+	Streams map[string]*Stream
+
+	Stream struct {
+		Labels map[string]string `json:"stream"`
+		Logs   []Log             `json:"values"`
+	}
+
+	Request struct {
+		Streams []Stream `json:"streams"`
+	}
+)
+
+func (s Streams) insertLog(ts []*telegraf.Tag, l Log) {
+	key := uniqKeyFromTagList(ts)
+	_, ok := s[key]
+
+	if !ok {
+		s[key] = newStream(ts)
+	}
```

Comment:
You could shorten this to a very common pattern:
```suggestion
	if _, ok := s[key]; !ok {
		s[key] = newStream(ts)
	}
```